code/game/machinery/recycle_vendor.dm

```diff
@@ -94,120 +84,208 @@
-	T.apply_to(merchants_pocket)
-
-	qdel(stored_item_object)
-	stored_item_object = null
-
-	for(var/i in stored_item_materials)
-		if(i in materials_supported)
-			if(i in materials_stored)
-				materials_stored[i] += stored_item_materials[i]
-			else
-				materials_stored.Add(i)
-				materials_stored[i] = stored_item_materials[i]
-
+	var/list/combinedmats = list()
+	for(var/obj/item/getthisone in stufftorecycle)
+		var/list/intermediary = getthisone.get_matter()
+		var/list/matter = intermediary?.Copy()
+		for(var/obj/O in getthisone.GetAllContents()) // can now recycle empty shells to get at the contents
+			if(length(O.get_matter()))
+				matter += O.get_matter()
+		for(var/toadd in matter)
+			combinedmats[toadd] += matter[toadd]
+		var/sellvalue = saleworthy_items[getthisone] * 0.8 // 20% fee
+		var/datum/transaction/T = new(-sellvalue, "", "Recycling payout for [getthisone.name]", src)
+		T.apply_to(silo.my_account)
+		qdel(getthisone) // we first add the item to the garbage queue
+
+	saleworthy_items.Cut(saleworthy_items.Find(stufftorecycle[1]),saleworthy_items.Find(length(stufftorecycle)))
+	stufftorecycle = null // then we remove what should be the remaining references
 	playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 50, 1)
-	spawn_money(stored_item_value, loc)
+	silo.addmaterial(combinedmats)
+	spawn_money(combinedvalue, loc)
 	flick("recycle_vend", src)
 	update_icon()
 
-	if(stored_item_value < 50 && prob(5))
+	if(combinedvalue < 50 && prob(10)) // selling less than around 32 sheets worth at a time is awfully small
 		speak("Ты бы еще консервных банок насобирал!")
 
+/obj/machinery/amerecycler/proc/recycle_and_output(obj/itemtorecycle)
+	if(!silo || silo.stat & NOPOWER)
+		flick("recycle_screen_red", overlays[1])
+		return
+	var/list/stufftorecycle = list()
+	if(itemtorecycle)
+		stufftorecycle.Add(itemtorecycle)
+	else
+		stufftorecycle |= saleworthy_items
+	if(!length(stufftorecycle))
+		return
+	var/combinedvalue = 0
+	for(var/currentitem in stufftorecycle)
+		combinedvalue += saleworthy_items[currentitem] / 5 // add all the fees together
+	if(combinedvalue > moneyinput) // if it's too much, cancel it all
+		flick("recycle_screen_red", overlays[1])
+		return FALSE
+	var/list/totalmaterials = list()
+	for(var/obj/currentitem in stufftorecycle)
+		var/currentitemsvalue = saleworthy_items[currentitem] // key is item, item value is associated with item
+		var/list/intermediary = currentitem.get_matter()
+		var/currentitemmaterials = intermediary?.Copy()
+		for(var/obj/O in currentitem.GetAllContents()) // can now recycle empty shells to get at the contents
+			if(length(O.get_matter()))
+				currentitemmaterials += O.get_matter()
+		for(var/materialtype in currentitemmaterials) // add them together so they can stack
+			var/amount = currentitemmaterials[materialtype]
+			totalmaterials[materialtype] += amount
+		currentitemsvalue *= 0.2 // 20% fee to recycle
```

Comment:
80% you mean?

Reply:
a "fee" is the amount of the value that the machine charges you to operate it.